GoogleForCreators/web-stories-wp
# Revamp and test build scripts

## Summary

Revamps the build tooling and makes it testable.

Adds a ton of unit tests for the build tooling, hence the many changed lines.

## Relevant Technical Choices

Bumps the project's node version to 14 and uses ECMAScript modules for node scripts, too.

Files still using CommonJS have been renamed to `cjs` because they don't yet work otherwise.

## Testing Instructions

```bash
./bin/commander.js --help

./bin/commander.js version --help

./bin/commander.js build-plugin --help

./bin/commander.js build-fonts --help
```

Verify that https://github.com/google/web-stories-wp/pull/1779/checks still lists ZIP files for the "Build plugin" workflow.

---



Fixes #1739
Fixes #1613
